tools/SourceKit/tools/swift-lang/CMakeLists.txt

```diff
@@ -4,7 +4,16 @@ if(NOT SWIFT_SOURCEKIT_USE_INPROC_LIBRARY AND SWIFT_BUILD_STDLIB)
   set(INSTALLED_COMP sourcekit-xpc-service)
   set(DEPENDS_LIST "sourcekitd-test")
 
-  add_swift_target_library(swiftSwiftLang SHARED
+  # The build type of swiftlang should agree with stdlib
+  set(SWIFT_LANG_LIBRARY_BUILD_TYPES)
+  if(SWIFT_BUILD_DYNAMIC_STDLIB)
+    list(APPEND SWIFT_LANG_LIBRARY_BUILD_TYPES SHARED)
+  endif()
+  if(SWIFT_BUILD_STATIC_STDLIB)
+    list(APPEND SWIFT_LANG_LIBRARY_BUILD_TYPES STATIC)
+  endif()
```

Comment:
Why is the build type of SourceKit associated to the standard library build type?  However, the initialisation here is pretty robust, so I would say that this is the preferable approach.

Reply:
Thank you for taking a look! @compnerd This is mostly because we'd like the way executable  finds stdlib also works for finding SwiftLang. Without this change, we sometimes need to additionally specify -L to teach an executable where to look for SwiftLang dylib.

Reply:
That really would be extremely helpful to have in a comment around there.